### PR TITLE
Document desc stanza

### PR DESCRIPTION
**Pull request by waldyrious**
Follow-up of https://github.com/Homebrew/brew/pull/8137, which implemented #77696.

Related threads, for future reference:
- #1790 (Add a description to cask)
- #3817 (App categories + descriptions)
- #5192 (Casks short description)
- #16089 (Formulae (info, search, etc) missing essential meta "description")

Note: The desc is currently added as an optional stanza. I've added descriptions to two casks (`alfred`, which is mentioned in the documentation, and `fontforge`, which was proposed as motivation in #77696). In https://github.com/Homebrew/brew/pull/8137, @core-code proposed to contribute additional descs. Beyond this, I would suggest also importing/sourcing descriptions from Wikidata ([example](https://www.wikidata.org/wiki/Q687332)) and Repology ([example](https://repology.org/project/fontforge/information)).

### CODE REVIEW

**Comment by waldyrious** [comment]
The CI failures (name and desc stanza out of order) will be fixed with https://github.com/Homebrew/brew/pull/8265. The order was encoded incorrectly in the previous PR, https://github.com/Homebrew/brew/pull/8137.

**Comment by waldyrious** [comment]
@reitermarkus are you able to restart the CI checks now that the brew PR has been merged?

File: doc/cask_language_reference/readme.md [hunk 1]
```diff
@@ -113,10 +112,9 @@ enclosed in single quotes.
 
 There are currently some arbitrary limitations on Cask tokens which are in the process of being removed. The Travis bot will catch any errors during the transition.
 
-
 ## Stanza order
 
-Having a common order for stanzas makes Casks easier to update and parse. Below is the complete stanza sequence (no Cask will have all stanzas). The empty lines shown here are also important, as they help to visually delineate information.
+Having a common order for stanzas makes Casks easier to update and parse. Below is the complete stanza sequence (no Cask will have all stanzas). The empty lines shown here are also important, as they help to visually delineate information. This is enforced by RuboCop, as configured [here](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/rubocops/cask/constants/stanza.rb).
```
Comment (reitermarkus) [inline doc/cask_language_reference/readme.md:117]: Remove this link, this is an implementation detail.

Comment (waldyrious) [reply doc/cask_language_reference/readme.md]: Sure — done.

**Comment by core-code** [comment]
> @core-code proposed to contribute additional descs. 

ok, i've attached some descriptions. quality is very varying, i guess some are fine while would need adapting or replacement.

note that we've written some of those ourselves - in this case you can consider them public domain. 
others are taken from description of the app on Mac App Store or from the description of the app from the homepage. in this case i guess they are copyrighted by the vendor. but i believe no vendor would have anything against using their description. its basically free promotion/marketing. also the copyrightability of such small text snippets could be disputed.

[descriptions.txt](https://github.com/Homebrew/homebrew-cask/files/5052191/descriptions.txt)

**Comment by waldyrious** [comment]
@core-code thanks a lot! I actually pinged you in this PR primarily for reference; I'd like to keep its diff relatively concise, and bulk-add descriptions in another PR. I'll take care of that separately if you don't mind. (Unless you'd like to submit them yourself — either way is fine by me!)

File: doc/cask_language_reference/all_stanzas.md [hunk 2]
```diff
@@ -4,57 +4,58 @@
...
-| `language`             | required                      | a Ruby block, called with language code parameters, containing other stanzas and/or a return value (see also [Language Stanza Details](stanzas/language.md))
-| `container nested:`    | no                            | relative path to an inner container that must be extracted before moving on with the installation; this allows us to support dmg inside tar, zip inside dmg, etc.
-| `container type:`      | no                            | a symbol to override container-type autodetect. May be one of: `:air`, `:bz2`, `:cab`, `:dmg`, `:generic_unar`, `:gzip`, `:otf`, `:pkg`, `:rar`, `:seven_zip`, `:sit`, `:tar`, `:ttf`, `:xar`, `:zip`, `:naked`. (example [parse.rb](https://github.com/Homebrew/homebrew-cask/blob/312ae841f1f1b2ec07f4d88b7dfdd7fbdf8d4f94/Casks/parse.rb#L11))
+| name                   | multiple occurrences allowed? | value               |
+| ---------------------- |------------------------------ | ------------------- |
+| `desc`                 | no                            | One-line description of the Cask. Shows when running `brew info`.<br />See [Desc Stanza Details](stanzas/desc.md) for more information.
```
Comment (reitermarkus) [inline doc/cask_language_reference/all_stanzas.md:47]: I think this should be a required stanza, even if we don't enforce it yet.

**Comment by core-code** [comment]
> and bulk-add descriptions in another PR. I'll take care of that separately if you don't mind. 

sounds good, please do

File: doc/cask_language_reference/all_stanzas.md (hunk 2 above)
Comment (waldyrious) [reply doc/cask_language_reference/all_stanzas.md:47]: Um, ok. I didn't want the documentation to be misleading, but yeah, I agree it's better to start telling people to treat this field as required as we work toward completion of the existing corpus.

Comment (waldyrious) [reply doc/cask_language_reference/all_stanzas.md]: @reitermarkus done!

**Comment by reitermarkus** [comment]
Thanks, @waldyrious!

**Pull request merged**